### DISCUSSION TITLE
Question: using TypeScript?

### DISCUSSION THREAD

**Discussion by gregwebs** [nytimes/library#261] [General]
Is nytimes/library interested in using TypeScript?

When a project takes on new developers and contributors, types can help dramatically improve confidence in contributions and reduce run-time errors.

This could require deployments to include an extra build step. I think heroku directly supports TypeScript, but for Google App Engine one needs to [include the build step in a Dockerfile](https://stackoverflow.com/questions/61586978/deploy-create-react-app-typescript-on-google-app-engine).

**Comment by afischer** [comment]
Hi @gregwebs,

Thanks for your suggestion! One of the goals when first building library was to make it easy for developers to contribute to the codebase regardless of their node experience and knowledge level. Because of this, we chose to avoid certain ES features and libraries with steeper learning curves in favor of tools already in use by a wide range of users across our newsroom.

The project has evolved significantly since then, so it's definitely worth having a conversations about introducing new dependencies to improve developer experience when working on the app.
